arkouda/pdarrayIO.py

```diff
@@ -633,15 +633,20 @@ def save_all(
             datasetNames = [str(column) for column in range(len(columns))]
     if (mode.lower() not in "append") and (mode.lower() not in "truncate"):
```

Comment:
I think the reason it was done this way was to allow mode to be `'a', 'app', 't',` instead of just `'append'` and `'truncate'`

That being said, i like this better. I think it's much cleaner and it matches the error message and the docstring